spyder_kernels/customize/namespace_manager.py

```diff
@@ -29,7 +41,7 @@ class NamespaceManager(object):
     def __init__(self, filename, namespace=None, current_namespace=False,
                  file_code=None):
         self.filename = filename
-        self.namespace = namespace
+        self.namespace = namespace, None
```

Comment:
Why do you need to add `None` here? That's really not clear to me.

Reply:
It seems you're using `self.namespace` to refer to `globals` and `locals`. If that's the case, please use `self.ns_globals` and `self.ns_locals` instead.